src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -47,7 +51,7 @@ public void AddWellKnownTypes()
         {
             var stringType = _typeSystemContext.GetWellKnownType(WellKnownType.String);
 
-            if (IsTypeInCompilationGroup(stringType))
+            if (IsInCompilationGroup(stringType))
             {
                 _rootProvider.AddCompilationRoot(stringType, "String type is always generated");
```

Comment:
The condition is just checking if System.String's assembly, System.Private.CoreLib is being compiled. If System.Private.CoreLib is part of compilation, System.String is generated.  I think the text still makes sense based on that, since it will always be available in the output binary.  It's only used in the dependency log for debugging anyhow.